File: src/main/java/kr/co/lunatalk/domain/image/repository/ImageRepository.java
package kr.co.lunatalk.domain.image.repository;

import kr.co.lunatalk.domain.image.domain.Image;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface ImageRepository extends JpaRepository<Image, Long>, ImageRepositoryCustom {
	Optional<Image> findByImageKey(String imageKey);
}

File: lunatalk-core/src/main/kotlin/kr/co/lunatalk/domain/image/repository/ImageRepository.kt
package kr.co.lunatalk.domain.image.repository

import kr.co.lunatalk.domain.image.domain.Image
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface ImageRepository : JpaRepository<Image, Long>, ImageRepositoryCustom {
    fun findByImageKey(imageKey: String): Optional<Image>
}
